src/inmanta/agent/handler.py

```diff
@@ -1067,3 +1067,83 @@ class HandlerNotAvailableException(Exception):
     This exception is thrown when a resource handler cannot perform its job. For example, the admin interface
     it connects to is not available.
     """
+
+
+class HandlerLogger(typing.Protocol):
```

Comment:
I think that if we port this to core we need to integrate it more tightly:
- I see no harm in this remaining a `Protocol` but I think we should explicitly inherit from it for both the actual handler's logger and the `LoggerWrapper` to make sure we stay compatible with the interface.
- if we lift `log_msg` to this class we can actually implement all of the others. It may then be cleaner to make this an `ABC` instead of a protocol with only `log_msg` an `abstractmethod`. This would simplify both `LoggerWrapper` and `HandlerContext`.
- Depending on how [this discussion](https://inmanta.slack.com/archives/CKRF0C8R3/p1681809378693359) goes we may be able to solve the issues you encountered with `exc_info` by refactoring `log_msg` a bit.
- The docstrings are written from the perspective of the module that used this. They should instead be written from the perspective of core.
- The name `LoggerWrapper` isn't very clear in this context. Then again I don't immediately have a better idea. `LoggingLogger`, `PlainLogger`, `SimpleLogger`, `NativeLogger`, `PythonLogger` etc come to mind but I don't know if I really like any of them.

Reply:
For the refactor, we could make it `_log_msg` here, then keep `HandlerContext.log_msg` for backwards compatibility to be on the safe side but add a deprecation warning. Once this is merged you can even immediately drop it from `master` (we diverged from iso6 last week).

Reply:
- Both of these classes should be marked as part of the stable API and included in the [docs](https://docs.inmanta.com/community/latest/model_developers/handlers.html#logging).

Reply:
I went with `IBasicLogging` for the interface and `KwargsLogger` for the wrapper wdyt ?

Reply:
The `I<something>` convention is not something we really use (I think it's more Java than Python but I'm not sure). We generally go for something like `<something>ABC` (from Abstract Base Class) if we want to explicitly indicate it's only an interface. I wouldn't say either is better than the other a priori, but I would definitely aim for consistency.

That aside:
- For the interface I prefer `Logger` over `Logging` because it's methods are instance methods so instances of the class are loggers. I'm not sure if `Basic` is required, could be just `Logger` / `LoggerABC` but I don't really care too much about that.
- `KwargsLogger` I would focus less on the kwargs and more on what/where it logs. It's a logger that just bridges the interface with Python's built-in logging. Words like Python, native, simple, ... would describe this better.